simple-loadflow/src/main/java/com/powsybl/loadflow/simple/equations/LoadFlowMatrix.java

```diff
@@ -71,105 +72,155 @@ private static Bus getBus2(Branch b) {
+
+            @Override
+            public void onBranch(int row, int column, Branch branch, Branch.Side side) {
+                if (!slackBusAdded) {
+                    if (column >= slackBusNum) {
+                        a.setValue(slackBusNum, slackBusNum, 1);
+                        slackBusAdded = true;
+                    }
+                }
+                if (row != slackBusNum && column != slackBusNum) {
+                    ClosedBranchDcFlowEquations eq = new ClosedBranchDcFlowEquations(branch);
+                    if (row == column) {
+                        if (side == Branch.Side.ONE) {
+                            a.addValue(row, column, eq.dp1dph1());
+                            rhs[row] -= eq.rhs1();
+                        } else {
+                            a.addValue(row, column, eq.dp2dph2());
+                            rhs[row] -= eq.rhs2();
+                        }
+                    } else {
+                        if (side == Branch.Side.ONE) {
+                            a.addValue(row, column, eq.dp1dph2());
+                        } else {
+                            a.addValue(row, column, eq.dp2dph1());
+                        }
+                    }
+                }
             }
-            if (rowP2 != 0) {
-                a.add(rowP2, colPh1, eq.dp2dph1());
-                a.add(rowP2, colPh2, eq.dp2dph2());
-            }
-        }
+        });
 
-        a.set(0, 0, 1);
+        stopwatch.stop();
+        LOGGER.info("DC matrix built in {} ms", stopwatch.elapsed(TimeUnit.MILLISECONDS));
 
         return a;
     }
 
-    public static PrimitiveDenseStore buildDcRhs(IndexedNetwork network) {
-        PrimitiveDenseStore rhs = PrimitiveDenseStore.FACTORY.makeZero(network.getBusCount(), 1);
+    public static double[] buildDcRhs(IndexedNetwork network, int slackBusNum) {
+        double[] rhs = new double[network.getBusCount()];
 
         for (Generator gen : network.get().getGenerators()) {
             Bus bus = gen.getTerminal().getBusView().getBus();
+            if (bus == null || !bus.isInMainConnectedComponent()) {
+                continue;
+            }
             int num = network.getIndex(bus);
-            if (num == 0) {
+            if (num == slackBusNum) {
                 continue;
             }
-            rhs.set(num, 0, gen.getTargetP());
+            rhs[num] -= gen.getTargetP();
         }
 
         for (Load load : network.get().getLoads()) {
             Bus bus = load.getTerminal().getBusView().getBus();
+            if (bus == null || !bus.isInMainConnectedComponent()) {
+                continue;
+            }
             int num = network.getIndex(bus);
-            if (num == 0) {
+            if (num == slackBusNum) {
                 continue;
             }
-            rhs.set(num, 0, load.getP0());
+            rhs[num] += load.getP0();
+        }
+
+        for (HvdcLine line : network.get().getHvdcLines()) {
```

Comment:
We'll need to update the javadoc to notify all the new features (HVDC, main component, PSTs working correctly ...)